### ISSUE TITLE
`SLF001` shouldn't be raised when accessing enum `_sunder_` methods

### ISSUE THREAD

**Issue by Bobronium** [astral-sh/ruff#11531]
**Keywords**: sunder, enum, magic, SLF001

`_sunder_` names are officially supported and mentioned in the docs, so they shouldn't be flagged as private: https://docs.python.org/3/library/enum.html#supported-sunder-names

### Reproduction
`$ cat test.py`
```py
from enum import Enum

Enum._missing_(1)
```

`$ ruff check test.py --isolated --select SLF001`
```shell
test.py:3:1: SLF001 Private member accessed: `_missing_`
Found 1 error.
```

`$ ruff --version`
```shell
ruff 0.4.5
````

**Comment by charliermarsh** [comment]
Closed by https://github.com/astral-sh/ruff/pull/11546.
